src/server/session.ts

```diff
@@ -1142,15 +1142,27 @@ namespace ts.server {
             return this.getPosition(args, scriptInfo);
         }
 
-        private getFileAndProject(args: protocol.FileRequestArgs): { file: NormalizedPath, project: Project } {
+        private getFileAndProject(args: protocol.FileRequestArgs): FileAndProject {
             return this.getFileAndProjectWorker(args.file, args.projectFileName);
         }
 
+        private getFileAndProjectForFileRename(args: protocol.GetEditsForFileRenameRequestArgs): FileAndProject {
+            const oldFilePath = toNormalizedPath(args.oldFilePath);
+            const oldProject = this.projectService.getDefaultProjectForFile(oldFilePath);
+            if (oldProject) return { file: oldFilePath, project: oldProject };
+
+            const newFilePath = toNormalizedPath(args.newFilePath);

```

Comment:
I don't know about the VS Code calling pattern, but VS will be making the request *after* the rename happens so I would expect the "new" code path to succeed much more often (i.e. might be worthwhile to check it first).